packages/kbn-discover-utils/src/components/custom_control_columns/degraded_docs_control.tsx

```diff
@@ -119,3 +126,44 @@ const DegradedDocs = ({
     />
   );
 };
+
+const EnableESQLDegradedDocsControl = ({
+  addIgnoredMetadataToQuery,
+}: Pick<DegradedDocsControlProps, 'addIgnoredMetadataToQuery'>) => {
+  const [isPopoverOpen, { off: closePopover, toggle: togglePopover }] = useBoolean(false);
+
+  return (
+    <EuiPopover
```

Comment:
Looks like the icon moves up/down depending on the state. Can we fix its visual alignment in regard to other row actions?

<img width="1151" alt="Screenshot 2024-10-11 at 10 25 23" src="https://github.com/user-attachments/assets/57939cff-5183-40d6-839b-8b135e5add7e">

(plus see the screenshots above)

Reply:
Good catch, there was a missing class that is generally applied to the control, [fix(discover): add missing css class](https://github.com/elastic/kibana/pull/195630/commits/15e6333133e2459ab380450a9b0658a1e3235fc9) fixes it.

Reply:
Ah, I see now. It should be using `Control` instead of `EuiButtonIcon`. Otherwise we might run into the issue with styles again or it would not render correctly if it happens to move under "More actions" menu in the future. Can use please replace it with `Control` usage?

Reply:
I didn't use the Control component because it imposes a tooltip wrapper around the button. Ideally this should be more composable and the tooltip should be decoupled from the Control component, but I didn't want to change its implementation since this was the first case, hence I directly used the EuiButtonIcon component in a different composition with the popover.

Reply:
We could change the implementation to make the tooltip optional or even allow a custom popover content for row actions in general.

Reply:
I updated the Control component to act as a normal EuiButtonIcon and conditionally add a tooltip when its specific `tooltipContent` property is passed. [refactor(discover): update Control component](https://github.com/elastic/kibana/pull/195630/commits/5c0a025168ff8e5c5b2c7b860ee5714e6b19a944)